src/spec.c

```diff
@@ -296,8 +279,8 @@ static void IndexSpec_TimedOutProc(RedisModuleCtx *ctx, WeakRef w_ref) {
 #endif
 }
 
-// Assuming the spec is properly locked before calling this function.
-// TODO: multithreaded: verify that the spec is locked
+// Assuming the GIL is held.
+// This can be done without locking the spec for write, since the timer is not modified or read by any other thread.
```

Comment:
This is called before we move to the main thread?

Reply:
this can be called from some background threads (like BG indexer) but for using the redis API for timers we need to have the GIL (and we currently do in all of the flows)
This makes the timer flag and timer id members of the spec guarded by the GIL and not the R/W lock, but it makes sense IMO